docs/docs/source-plugin-tutorial.md

```diff
@@ -0,0 +1,393 @@
+  { boundActionCreators, createNodeId },
+  configOptions
+) => {
+  const { createNode } = boundActionCreators
+
+  // Gatsby adds a configOption that's not needed for this plugin, delete it
+  delete configOptions.plugins
+
+  // Helper function that processes a photo to match Gatsby's node structure
+  const processPhoto = photo => {
+    const nodeId = createNodeId(`pixabay-photo-${photo.id}`)
+    const nodeContent = JSON.stringify(photo)
+    const nodeContentDigest = crypto
+      .createHash('md5')
+      .update(nodeContent)
+      .digest('hex')
+
+    const nodeData = Object.assign({}, photo, {
+      id: nodeId,
+      parent: null,
+      children: [],
+      internal: {
+        type: `PixabayPhoto`,
+        content: nodeContent,
+        contentDigest: nodeContentDigest,
+      },
+    })
+
+    return nodeData
+  }
+
+  // Convert the options object into a query string
+  const apiOptions = queryString.stringify(configOptions)
+
+  // Join apiOptions with the Pixabay API URL
+  const apiUrl = `https://pixabay.com/api/?${apiOptions}`
+
+  // Gatsby expects sourceNodes to return a promise
+  return (
+    // Fetch a response from the apiUrl
+    fetch(apiUrl)
+      // Parse the response as JSON
+      .then(response => response.json())
+      // Process the response data into a node
+      .then(data => {
+        // For each query result (or 'hit')
+        data.hits.forEach(photo => {
+          // Process the photo data to match the structure of a Gatsby node
+          const nodeData = processPhoto(photo)
+          // Use Gatsby's createNode helper to create a node from the node data
+          createNode(nodeData)
+        })
+      })
+  )
+}
+```
+
+### Query for results
+
+Your plugin is ready. Restart `gatsby dev` and open a browser at [http://localhost:8000/\_\_\_graphql](http://localhost:8000/___graphql). The Pixabay data can be queried from here, try:
+
+```graphql
+{
+  allPixabayPhoto(limit: 10) {
+    edges {
+      node {
+        largeImageURL
+        pageURL
+        tags
+        user
+      }
+    }
+  }
+}
+```
+
+Or [open the query from this link](<http://localhost:8000/___graphql?query=%7B%0A%20%20allPixabayPhoto(limit%3A%2010)%20%7B%0A%20%20%20%20edges%20%7B%0A%20%20%20%20%20%20node%20%7B%0A%20%20%20%20%20%20%20%20largeImageURL%0A%20%20%20%20%20%20%20%20pageURL%0A%20%20%20%20%20%20%20%20tags%0A%20%20%20%20%20%20%20%20user%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A>).
+
+Experiment with different options in your `gatsby-config.js` file to see how that affects your query results. The [Pixabay API docs](https://pixabay.com/api/docs/#api_search_images) might be a useful reference.
+
+## Publishing a plugin
+
+You've built a local plugin for your project, but what if you want to share it with others? `npm` allows you to publish your plugins, check out the npm docs on [How to Publish & Update a Package](https://docs.npmjs.com/getting-started/publishing-npm-packages) for more info.
```

Comment:
*plugins. Check out...

Reply:
Yeah I'm still a bit undecided about how much detail to go into here. The plugin authoring doc is mentioned later on so I skipped it here. And I wondered if mentioning the plugin library here might lead to a bunch of duplicate `gatsby-source-pixabay` plugins being added. 

What do you think? Maybe it's worth mentioning the plugin library in the _What's next?_ section? Although if someone is interested in writing their own plugin, can we assume they're already aware of the plugin library?